docs/docs/evaluation/configuration-reference.md

```diff
@@ -0,0 +1,227 @@
+
+## API Reference
+
+For a full API reference of all modules exported by `mongodb-chatbot-evaluation`, refer to the [API Reference](../reference/evaluation/) documentation.
+
+This page links to the key reference documentation for configuring the Ingest CLI.
+
+## `ConfigConstructor`
+
+The [`ConfigConstructor`](../reference/evaluation/modules.md#configconstructor) function is the root configuration type for the Ingest CLI. This exports an [`EvalConfig`](../reference/evaluation/interfaces/EvalConfig.md) object.
+
+## Data Stores
+
+### `CommandMetadataStore`
+
+The [`CommandMetadataStore`](../reference/evaluation/interfaces/CommandMetadataStore.md) is an interface to interact with MongoDB collection that tracks metadata of each command run.
+
+To create an `CommandMetadataStore`, use the function [`makeMongoDbCommandMetadataStore()`](../reference/evaluation/modules.md#makemongodbcommandmetadatastore).
+
+```ts
+import { makeMongoDbCommandMetadataStore } from "mongodb-chatbot-evaluation";
+
+const commandMetadataStore = makeMongoDbCommandMetadataStore({
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+### `GeneratedDataStore`
+
+The [`GeneratedDataStore`](../reference/evaluation/interfaces/GeneratedDataStore.md) is an interface to interact with MongoDB collection that stores generated data.
+
+To create an `GeneratedDataStore`, use the function [`makeMongoDbGeneratedDataStore()`](../reference/evaluation/modules.md#makemongodbgenerateddatastore).
+
+```ts
+import { makeMongoDbGeneratedDataStore } from "mongodb-chatbot-evaluation";
+
+const generatedDataStore = makeMongoDbGeneratedDataStore({
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+### `EvaluationStore`
+
+The [`EvaluationStore`](../reference/evaluation/interfaces/EvaluationStore.md) is an interface to interact with MongoDB collection that stores evaluation data.
+
+To create an `EvaluationStore`, use the function [`makeMongoDbEvaluationStore()`](../reference/evaluation/modules.md#makemongodbevaluationstore).
+
+```ts
+import { makeMongoDbEvaluationStore } from "mongodb-chatbot-evaluation";
+
+const evaluationStore = makeMongoDbEvaluationStore({
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+### `ReportStore`
+
+The [`ReportStore`](../reference/evaluation/interfaces/ReportStore.md) is an interface to interact with MongoDB collection that stores reports.
+
+To create an `ReportStore`, use the function [`makeMongoDbReportStore()`](../reference/evaluation/modules.md#makemongodbreportstore).
+
+```ts
+import { makeMongoDbReportStore } from "mongodb-chatbot-evaluation";
+
+const reportStore = makeMongoDbReportStore({
+  connectionUri: MONGODB_CONNECTION_URI,
+  databaseName: MONGODB_DATABASE_NAME,
+});
+```
+
+## Test Cases
+
+You must provide test cases to evaluate the chatbot. Pass the test cases to the `commands.generate` property in the `EvalConfig`.
```

Comment:
````suggestion
You must provide test cases to evaluate the chatbot. Pass the test cases to the `commands.generate` property in the `EvalConfig`.

```typescript
const testCases: ConversationTestCase[] = [
  {
    name: `It understands "why the chicken crossed the road" jokes`,
    expectation: `
      The ASSISTANT responds with a completion of the classic chicken crossing the road joke.
      The joke should be completed in a way that is both humorous and appropriate.
    `,
    tags: ["joke"],
    messages: [
      { role: "user", content: "Why did the chicken cross the road?" }
    ]
  },
];

const evalConfig: EvalConfig = {
  // ... other fields,
  commands: {
    generate: {
      myTest: {
        type: "conversation",
        testCases: testCases,
        generator: makeGenerateConversationData({ ... }),
      },
    },
    evaluate: { /* ... */ },
    report: { /* ... */ },
  },
};
```
````

I think this benefits from an up front example of what they look like / where they're defined.